### ISSUE TITLE
10.000+ jobs in queue after update to Craft 4 and latest Blitz Cache

### ISSUE THREAD

**Issue by sander-lameco** [putyourlightson/craft-blitz#462]
4 hours ago we deployed a new version to a live website with Craft 4 and latest Blitz Cache.
Blitz Cache started generating queue jobs for `Generating Blitz cache` and after watching it a while jobs started going up. Now after 4 hours the number of jobs is already up to 10.570 jobs.

We can see about ~25 `Generating Blitz cache` jobs of which 8 seem to be running and other pending.

We are using the `Expire the cache, regenerate manually or organically` setting and have a cronjob set up for `5 * * * *`.

Server (CPU) load is really high as well. Is this normal behaviour?

**Comment by bencroker** [comment]
This is _not_ normal behaviour, unless a task was initiated to resave over 10k elements. If that's the case, or if the jobs are still running then I'd suggest you disable Blitz caching, clear all queue jobs, and then enable Blitz caching and refresh the cache. Normally, resaving elements in batches resulting in `batchMode` being to `true`, which tells Blitz to only create a queue job after the batch has completed. If you a custom plugin/module is initiating the resaves then this may have to be done manually, see the [docs on integrations](https://putyourlightson.com/plugins/blitz#integrations).

**Comment by sander-lameco** [comment]
Well it's a pretty big website with over 1500+ entries across 10 sites. After the Blitz cache jobs there we're a lot of smaller jobs for resaving elements.

Probably it took more then an hour for the Blitz Cache to generate and the cronjob triggered the cache before it ever finished to generate them all.

**Comment by bencroker** [comment]
Did you figure out what caused the element resaves which triggered the Blitz refresh jobs? If you do then I can try to prevent that from happening again and to others.

**Comment by sander-lameco** [comment]
Not yet, we will do some testing to find out what caused it to hopefully prevent it from happening next time. I will let you know if we find anything. Should we keep this issue open till then?

**Comment by bencroker** [comment]
Yes, thanks.

**Comment by sander-lameco** [comment]
So I've been doing some local testing and one thing I notice is that the `Generating Blitz Cache` job and some others as well are in there multiple times (3 times) after just changing a single entries title. (see attachments) The weird part is, it is not the amount of sites we have(10 sites of which 9 enabled). 

**Utilities -> Blitz Cache**
![Screenshot 2023-02-01 at 15 13 58](https://user-images.githubusercontent.com/25500086/216067628-01da2ec1-9d81-4de3-967c-13928107bbf6.png)

**Utilities -> Queue Manager** (1 `Generating Blitz Cache` was already finished at the moment of screenshot)
![Screenshot 2023-02-01 at 15 16 47](https://user-images.githubusercontent.com/25500086/216067640-cd5f46c4-d5cb-457d-bd9b-2f7251dab854.png)

I tried using
``` twig
{% do craft.blitz.options.cacheElements(false).cacheElementQueries(false) %}
```

To disable the parts of the header and footer which uses verbb/navigation to see if that made any difference but it didn't. What are we missing?

**Comment by sander-lameco** [comment]
I have been doing some extra testing and we are curious as of how the amount of pages is being determined?

We did a test with 2 sites within the same Craft environment with each 38 pages cached.
After changing the title of the page in one site it started generating new cache.
We would expect the job to say `Generating X of 38` but instead it said `Generating X of 90`. 
The title is translatable for each site.

**Comment by bencroker** [comment]
There are possibly variations of pages using query strings. If you look in the `blitz_caches` database table then you'll see a full list of all pages that Blitz is tracking. If you want to refresh it then you can do so by refreshing the cache.

**Comment by sander-lameco** [comment]
Thanks for the reply.

So I looked into the table and filtered on siteId for one of the cached sites. They seem to match: 38 pages cached in the Blitz Cache overview and 38 when filtering on that siteId in the DB table.

**Comment by bencroker** [comment]
If that entry is enabled for 2 sites, then updating it may trigger refreshes for both sites. Blitz 4.3.0 is close to being released and will contain some nice ways of reducing how many pages must be regenerated, I'll let you know once it is released.

**Comment by sander-lameco** [comment]
Yes that is the case, it is a multi(locale)site with around 9 sites sharing the majority of pages (~200) between these sites. Looking forward to these new features in 4.3.0.

**Comment by bencroker** [comment]
Just released version 4.3.0. Read about the new [includeCached](https://putyourlightson.com/articles/blitz-4-3-feature-release) function.
